### ISSUE TITLE
Access to app_metadata and mapping to Authorities

### ISSUE THREAD

**Issue by nealeu** [auth0/auth0-spring-security-api#33]
Hi,

In https://auth0.com/docs/tutorials/using-auth0-with-multi-tenant-apps you talk about using app_metadata for managing permissions.

We'd like to do this, but this library provides us no path to be able to provide configuration to do this.

As far as I can see we need to be able to provide a strategy into the innards of AuthenticationJsonWebToken to map app_metadata and expose the authorities.

This requires a fair degree of 'wiring' refactoring.  Are you open to this approach?

**Comment by nealeu** [comment]
@nikolaseu Any chance of a response on this?  

Having now switched to Spring Security OAuth2 and patched that (it's easier as the beans can be overridden), I'm puzzled as to why you are doing your own version instead of working with Spring/Pivotal.

**Comment by KarmicJohn** [comment]
@nealeu - do you have an example of what you've overrode here to get access to the app_metadata for adding authorities to the access filters?

**Comment by skliche** [comment]
Yup we also switched to Spring Security OAuth2 and removed all auth0 dependencies. To map special claims of the token, I wrote a UserAuthenticationConverter and injected my implementation in the DefaultAccessTokenConverter. Hope this helps.

**Comment by bastman** [comment]
We have the same issue. Our authorities are stored in "scope" and "some-prefix/claims/roles". Is there any chance to make them accessible?

**Comment by skliche** [comment]
Is Spring Security OAuth2 not an option for you?
